pilot/pkg/config/file/store.go

```diff
@@ -349,61 +347,39 @@ func (s *KubeSource) parseChunk(r *collection.Schemas, name string, lineNum int,
 		return resources, nil
 	}
 
-	if groupVersionKind.Empty() {
-		return resources, fmt.Errorf("unable to parse resource with no group, version and kind")
+	objects, err := object.ParseK8sObjectsFromYAMLManifestFailOption(string(yamlChunk), true)
```

Comment:
Revise analyze parsing to object parsing, there's no behavior change except the error message being from `unable to parse resource with no group, version and kind` to `unable to parse invalid k8s object`